### PR TITLE
Add a hash_resources parameter (column) to the macOS signature table

### PR DESCRIPTION
**Pull request by alessandrogario**
This is an attempt at solving issue #2987 

The new column allows the user to disable resource hashing when querying the table.

```
Using a virtual database. Need help, type '.help'
osquery> SELECT *, hash_resources FROM signature WHERE path = '/Applications/Xcode.app' AND hash_resources = 0;
+-------------------------+--------+--------------------+------------------------------------------+-----------------+----------------------------------+----------------+
| path                    | signed | identifier         | cdhash                                   | team_identifier | authority                        | hash_resources |
+-------------------------+--------+--------------------+------------------------------------------+-----------------+----------------------------------+----------------+
| /Applications/Xcode.app | 1      | com.apple.dt.Xcode | 73a18b1658fad362c2e13359feafc517b1325870 | 59GAB85EFG      | Apple Mac OS Application Signing | 0              |
+-------------------------+--------+--------------------+------------------------------------------+-----------------+----------------------------------+----------------+
```

EDIT: Got rid of a static variable I didn't initially notice in the code! Everything works fine now!

### CODE REVIEW

**Comment by terracatta** [comment]
This is awesome. The caching is weird, I hope someone can point us to why that is happening.

In an ideal world would we want` hash_resources` to default to 0? I know  defaulting to "1" is the right thing to do for backwards compatibility, but since this change will likely be debuted in osquery 3, this is an opportunity to make a backwards incompatible change. 

Just food for thought.

File: osquery/tables/system/darwin/signature.mm [hunk 1]
```diff
@@ -26,55 +26,55 @@
...
+  } catch (const bad_lexical_cast& e) {
+    VLOG(1) << "Bad OSX minor version returned by the os_version table";
+    return -1;
+  }
+}
+
+// Get the flags to pass to SecStaticCodeCheckValidityWithErrors, depending on
+// the OS version.
+SecCSFlags getVerifyFlags(bool hash_resources) {
+  static int osMinorVersion = getOSMinorVersion();
```
Comment (muffins) [inline osquery/tables/system/darwin/signature.mm]: nit: does this have to be static? What about `auto`?

File: osquery/tables/system/darwin/signature.mm [hunk 2]
```diff
@@ -26,55 +26,55 @@
...
 // Generate a signature for a single file.
-void genSignatureForFile(const std::string& path, QueryData& results) {
+void genSignatureForFile(const std::string& path,
+                         QueryData& results,
+                         bool hash_resources) {
   Row r;
   OSStatus result;
 
   // Defaults
+  r["hash_resources"] = (hash_resources ? "1" : "0");
```
Comment (muffins) [inline osquery/tables/system/darwin/signature.mm]: Hrm, seems like extra, but I'd use the `INTEGER` macro with the numeric constants.

**Comment by muffins** [comment]
ok to test

**Comment by muffins** [comment]
Ah, also looks like this will need to be rebased, we're so slow T.T

**Comment by zwass** [comment]
@alessandrogario @muffins can you guys get this rebased and merged?

**Comment by alessandrogario** [comment]
Hello @zwass! I've updated the pull requests, can you have a look at it? Thanks!

**Comment by zwass** [comment]
It looks good to me. Maybe @muffins or someone else from FB can take a look?

**Pull request merged**